sdk/src/main/java/com/mapbox/maps/MapController.kt

```diff
@@ -123,13 +123,13 @@ internal class MapController : MapPluginProviderDelegate, MapControllable {
       addOnStyleDataLoadedListener(onStyleDataLoadedListener)
     }
     renderer.onStart()
-    pluginRegistry.onStart()
     if (!mapboxMap.isStyleLoadInitiated) {
       // Load the style in mapInitOptions if no style has loaded yet.
       mapInitOptions.styleUri?.let {
         mapboxMap.loadStyleUri(it)
       }
     }
+    pluginRegistry.onStart()
```

Comment:
On internal convo, you mentioned:
```
What if we will change here:
    pluginRegistry.onStart()
    if (!mapboxMap.isStyleLoadInitiated) {
      // Load the style in mapInitOptions if no style has loaded yet.
      mapInitOptions.styleUri?.let {
        mapboxMap.loadStyleUri(it)
      }
    }
to do like this:
    if (!mapboxMap.isStyleLoadInitiated) {
      // Load the style in mapInitOptions if no style has loaded yet.
      mapInitOptions.styleUri?.let {
        mapboxMap.loadStyleUri(it)
        pluginRegistry.onStart()
      }
    }
```

Proposed change here is actually not part of the style loading callback?

Reply:
I think we should call `onStart` inside the Activity's `onStart` event, regardless of the style loading status. 
and I remember we do have a `onStyleChanged` listener, we could hook to this listener inside the plugin if the style is need for certain action, wdyt?

Reply:
@pengdev you are right, this `onStart` is initially called from Activity/Fragment and implementation is inside *MapController*, It is entry point.

Reply:
> Proposed change here is actually not part of the style loading callback?


I think the point of this change is that the new style load will clear the style load listeners that previous `getStyle() { style -> } ` sets. 
In some plugins we rely on the async `getStyle()` call to do the initialisation in the plugin's onStart(). And when the new style is loaded after the plugin's onStart, it will result in the plugins' getStyle() callback never being fired.
This fix intentionally moved the pluginRegistry.onStart() after the `loadStyleUri` call, so that we made sure the styleLoaded listener set by plugins' getStyle() is not cleared.